.github/actions/update-axe-core-v1/src/run.ts

```diff
@@ -0,0 +1,104 @@
+import type { Core } from './types'
+import { glob, type GlobOptions } from 'glob'
+import { getExecOutput } from '@actions/exec'
+import path from 'path'
+import type { PackageManager } from './types'
+
+export default async function run(
+  core: Core,
+  getPackageManager: (dirPath: string) => PackageManager,
+  cwd?: string
+) {
+  try {
+    const { stdout: latestAxeCoveVersion } = await getExecOutput('npm', ['info', 'axe-core', 'version'])
+    core.info(`latest axe-core version ${latestAxeCoveVersion}`)
+
+    // npm and yarn workspaces will have a lock file at the root
+    // of the project
+    const rootPackageManager = await getPackageManager('./')
+    core.info(`root package manager detected as ${rootPackageManager}`)
+
+    const packages = await glob('**/package.json', {
+      absolute: true,
+      cwd,
+      ignore: '**/node_modules/**/package.json'
+    })
+    let installedAxeCoreVersion: string | null = null
+
+    for (const filePath of packages) {
+      core.info(`package.json found in ${filePath}`)
+      const pkg = await import(filePath as string)
+
+      if (
+        !pkg.dependencies?.['axe-core'] &&
+        !pkg.devDependencies?.['axe-core']
+      ) {
+        core.info(`no axe-core dependency`)
+        continue;
+      }
+
+      const dirPath = path.dirname(filePath as string)
+      const packageManager = await getPackageManager(dirPath) ?? rootPackageManager ?? 'npm'
```

Comment:
Do we want to default to `npm` here, instead of handling `undefined`? 

```suggestion
      const packageManager = await getPackageManager(dirPath) ?? rootPackageManager
      
      if (!packageManager) {
      	core.warning('Unsupported package manager, moving on....'): 
      	continue
      }
```

Reply:
We could get into the case where a package prevents the `package-lock.json` creation through an `.nvmrc` file. I think we can assume that `npm` is a safe default if we can't detect it from a file.

Reply:
I agree with Gabe, though I'd actually go even stronger and consider it an immediate fatal error if we hit a case where we can't reliably determine the package manager, not just a warning. I would much rather fail fast than accept any unnecessary risk of a bot making invalid updates with customer impact.